erpnext/healthcare/doctype/patient_encounter/patient_encounter.py

```diff
@@ -33,6 +34,100 @@ def set_title(self):
 
+	@frappe.whitelist()
+	@staticmethod
+	def get_applicable_treatment_plans(encounter):
+		patient = frappe.get_doc('Patient', encounter['patient'])
+
+		plans = None
+		plan_filters = {}
+		plan_filters['name'] = ['in', []]
+
+		age = patient.age
+		if age:
+			plan_filters['patient_age_from'] = ['>=', age.years]
+			plan_filters['patient_age_to'] = ['<=', age.years]
+
+		gender = patient.sex
+		if gender:
+			plan_filters['gender'] = gender
+
+		diagnosis = [_diagnosis['diagnosis'] for _diagnosis in encounter['diagnosis']]
+		if diagnosis:
+			filters = [
+				['diagnosis', 'in', diagnosis],
+				['parenttype', '=', 'Treatment Plan Template'],
+			]
+			diagnosis = frappe.get_list('Patient Encounter Diagnosis', filters=filters, fields='*')
+			plan_names = [_diagnosis['parent'] for _diagnosis in diagnosis]
+			plan_filters['name'][1].extend(plan_names)
+
+		symptoms = [symptom['complaint'] for symptom in encounter['symptoms']]
+		if symptoms:
+			filters = [
+				['complaint', 'in', symptoms],
+				['parenttype', '=', 'Treatment Plan Template'],
+			]
+			symptoms = frappe.get_list('Patient Encounter Symptom', filters=filters, fields='*')
+			plan_names = [symptom['parent'] for symptom in symptoms]
+			plan_filters['name'][1].extend(plan_names)
+
+		if not plan_filters['name'][1]:
+			plan_filters.pop('name')
+
+		plans = frappe.get_list('Treatment Plan Template', fields='*', filters=plan_filters)
+
+		return plans
+
+	@frappe.whitelist()
+	def fill_treatment_plans(self, treatment_plans=None):
+		for treatment_plan in treatment_plans:
+			self.fill_treatment_plan(treatment_plan)
+
+	def fill_treatment_plan(self, plan):
+		plan_items = frappe.get_list('Treatment Plan Template Item', filters={'parent': plan}, fields='*')
+		for plan_item in plan_items:
+			self.fill_treatment_plan_item(plan_item)
+
+		drugs = frappe.get_list('Drug Prescription', filters={'parent': plan}, fields='*')
+		for drug in drugs:
+			self.fill_treatment_plan_drug(drug)
+
+	def fill_treatment_plan_drug(self, drug=None):
+		doc = frappe.new_doc('Drug Prescription')
+		doc.drug_code = drug['drug_code']
+		doc.dosage = drug['dosage']
+		doc.dosage_form = drug['dosage_form']
+		doc.period = drug['period']
+		doc.parenttype = drug['parenttype']
+		doc.parent = drug['parent']
+		doc.insert()
+		self.append('drug_prescription', doc)
+		self.save()
+
+	def fill_treatment_plan_item(self, plan_item):
+		if plan_item['type'] == 'Clinical Procedure Template':
+			doc = frappe.new_doc('Procedure Prescription')
+			doc.procedure = plan_item['template']
+			child_field = 'procedure_prescription'
+
+		if plan_item['type'] == 'Lab Test Template':
+			doc = frappe.new_doc('Lab Prescription')
+			doc.lab_test_code = plan_item['template']
+			child_field = 'lab_test_prescription'
+
+		if plan_item['type'] == 'Therapy Type':
+			doc = frappe.new_doc('Therapy Plan Detail')
+			doc.therapy_type = plan_item['template']
+			child_field = 'therapies'
```

Comment:
This change is remaining: https://github.com/frappe/erpnext/pull/26557#discussion_r690053005